### PR TITLE
glances 2.11.1 (new formula)

### PR DESCRIPTION
**Pull request by alyssais**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Guessing I'm not the only one missing htop on High Sierra!

### CODE REVIEW

File: Formula/glances.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    begin
+      read, write = IO.pipe
+      pid = spawn(bin/"glances", "-q", "--export-csv", "/dev/stdout", out: write)
```
Comment (ilovezfs) [inline Formula/glances.rb]: fork/exec

File: Formula/glances.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  url "https://github.com/nicolargo/glances/archive/v2.11.1.tar.gz"
+  sha256 "446a0ee6e13c0c7ceb4bc0d4868add8a02d5e3ff866de8e880bdb33dce6ab3fc"
+
+  resource "psutil" do
+    url "https://files.pythonhosted.org/packages/e2/e1/600326635f97fee89bf8426fef14c5c29f4849c79f68fd79f433d8c1bd96/psutil-5.4.3.tar.gz"
+    sha256 "e2467e9312c2fa191687b89ff4bc2ad8843be4af6fb4dc95a7cc5f7d7a327b18"
+  end
+
+  def install
+    virtualenv_install_with_resources
```
Comment (ilovezfs) [inline Formula/glances.rb]: let's avoid virtualenv. see snakemake

File: Formula/glances.rb (hunk 1 above)
Comment (alyssais) [reply Formula/glances.rb]: Curious: why?

Comment (ilovezfs) [reply Formula/glances.rb]: consistency.

File: Formula/glances.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    begin
+      read, write = IO.pipe
+      pid = fork do
+        exec bin/"glances", "-q", "--export-csv", "/dev/stdout", :out => write
```
Comment (alyssais) [inline Formula/glances.rb]: Note that this test currently hits the network to determine public IP. There's a `--disable-ip` flag but it doesn't actually prevent the network request. I've opened a [pull request](https://github.com/nicolargo/glances/pull/1216) to fix this.

Comment (ilovezfs) [reply Formula/glances.rb]: hitting the network is fine

File: Formula/glances.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  resource "psutil" do
+    url "https://files.pythonhosted.org/packages/e2/e1/600326635f97fee89bf8426fef14c5c29f4849c79f68fd79f433d8c1bd96/psutil-5.4.3.tar.gz"
+    sha256 "e2467e9312c2fa191687b89ff4bc2ad8843be4af6fb4dc95a7cc5f7d7a327b18"
+  end
+
+  depends_on :python
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+    resources.each do |r|
```
Comment (ilovezfs) [inline Formula/glances.rb]: there's only one so `resource("psutil")` to avoid the loop

File: Formula/glances.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  homepage "https://nicolargo.github.io/glances/"
+  url "https://github.com/nicolargo/glances/archive/v2.11.1.tar.gz"
+  sha256 "446a0ee6e13c0c7ceb4bc0d4868add8a02d5e3ff866de8e880bdb33dce6ab3fc"
+
+  resource "psutil" do
+    url "https://files.pythonhosted.org/packages/e2/e1/600326635f97fee89bf8426fef14c5c29f4849c79f68fd79f433d8c1bd96/psutil-5.4.3.tar.gz"
+    sha256 "e2467e9312c2fa191687b89ff4bc2ad8843be4af6fb4dc95a7cc5f7d7a327b18"
+  end
+
+  depends_on :python
```
Comment (ilovezfs) [inline Formula/glances.rb]: `depends_on "python" if MacOS.version <= :snow_leopard`

File: Formula/glances.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
+    resources.each do |r|
+      r.stage do
+        system "python2.7", *Language::Python.setup_install_args(libexec/"vendor")
```
Comment (ilovezfs) [inline Formula/glances.rb]: "python"

File: Formula/glances.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      end
+    end
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python2.7", *Language::Python.setup_install_args(libexec)
```
Comment (ilovezfs) [inline Formula/glances.rb]: "python"

File: Formula/glances.rb [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  test do
+    begin
+      read, write = IO.pipe
+      pid = fork do
+        exec bin/"glances", "-q", "--export-csv", "/dev/stdout", :out => write
+      end
+      header = read.gets
+      assert_match(/timestamp/, header)
+    ensure
+      Process.kill(:TERM, pid)
```
Comment (ilovezfs) [inline Formula/glances.rb]: `Process.kill("TERM", pid)`

File: Formula/glances.rb [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  test do
+    begin
+      read, write = IO.pipe
+      pid = fork do
+        exec bin/"glances", "-q", "--export-csv", "/dev/stdout", :out => write
+      end
+      header = read.gets
+      assert_match(/timestamp/, header)
```
Comment (ilovezfs) [inline Formula/glances.rb]:
```
assert_match "timestamp", header
```

File: Formula/glances.rb [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  test do
+    begin
+      read, write = IO.pipe
+      pid = fork do
+        exec bin/"glances", "-q", "--export-csv", "/dev/stdout", :out => write
+      end
+      header = read.gets
+      assert_match /timestamp/, header
```
Comment (ilovezfs) [inline Formula/glances.rb]: would just use "timestamp" here. it converts to regex via magic

File: Formula/glances.rb [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  homepage "https://nicolargo.github.io/glances/"
+  url "https://github.com/nicolargo/glances/archive/v2.11.1.tar.gz"
+  sha256 "446a0ee6e13c0c7ceb4bc0d4868add8a02d5e3ff866de8e880bdb33dce6ab3fc"
+
+  resource "psutil" do
+    url "https://files.pythonhosted.org/packages/e2/e1/600326635f97fee89bf8426fef14c5c29f4849c79f68fd79f433d8c1bd96/psutil-5.4.3.tar.gz"
+    sha256 "e2467e9312c2fa191687b89ff4bc2ad8843be4af6fb4dc95a7cc5f7d7a327b18"
+  end
+
+  depends_on "python" if MacOS.version <= :snow_leopard
```
Comment (ilovezfs) [inline Formula/glances.rb]: should be above the resource

**Comment by ilovezfs** [comment]
man page is getting installed in
```
/usr/local/Cellar/glances/2.11.1/libexec/share/man/man1/glances.1
```
so probably should jettison that over to man1

**Comment by ilovezfs** [comment]
The man page specifically mentions
```
macOS        | ~/Library/Application Support/glances, /usr/local/etc/glances 
```
for config file. You may want o make sure that works for other prefixes too not just /usr/local.

**Comment by ilovezfs** [comment]
```
/usr/local/Cellar/glances//2.11.1/libexec/share/doc/glances/glances.conf
```
That should probably be in pkgshare

**Comment by ilovezfs** [comment]
or doc

or etc renamed to .example

**Comment by ilovezfs** [comment]
man page says to expect it

```
          A template is available in the /usr{,/local}/share/doc/glances (Unix-like) directory or directly on GitHub.
```

So that would be `doc`

**Comment by ilovezfs** [comment]
If we don't think people are going to care about using the config file, I'm happy to look the other way on the non-standard prefix stuff, but we should at least get the man page installed.

**Review by ilovezfs — APPROVED** [review]

**Comment by alyssais** [comment]
> If we don't think people are going to care about using the config file, I'm happy to look the other way on the non-standard prefix stuff

👍 for leaving that as a future enhancement.

**Comment by ilovezfs** [comment]
Shipped! Thanks for the :new: formula @alyssais.

>👍 for leaving that as a future enhancement.

Yeah, I think someone who actually wants to use the config file in a non-standard prefix can take that up.

**Pull request closed without merging**